src/lib/x509/x509_ext.cpp

```diff
@@ -764,6 +768,39 @@ void CRL_Issuing_Distribution_Point::decode_inner(const std::vector<uint8_t>& bu
    BER_Decoder(buf).decode(m_distribution_point).verify_end();
 }
 
+void TNAuthList::TNEntry::encode_into(DER_Encoder&) const {}
+
+void TNAuthList::TNEntry::decode_from(class BER_Decoder& ber) {
+   BER_Object obj = ber.get_next_object();
+
+   type = static_cast<decltype(type)>(obj.type_tag());
```

Comment:
Casting the `type_tag()` looks a bit funky to me. Admittedly though, my knowledge about the `BER_Decoder` is somewhat limited. @randombit please have a look. :)

Reply:
`type_tag()` contains enum value here because of the explicit context-specific tagging. on the out-of-range values it would cast without errors anyway and throw exception in the default switch case below.